### PR TITLE
fixed bot.checkRole method

### PR DESCRIPTION
**Pull request by sergeme**
rolechecker.js: moved bot.checkRole from bot.js to constants/rolechecker.js. checking roles works now. returns only true if user is in at least on of the required roles. console output for logging purposes.

xp.js & reboot.js: bot.checkRole call fixed with await. granting xp works, console still throws some errors which have to do with level.js.

please add to master.

### CODE REVIEW

File: src/constants/rolechecker.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+const bot = require('../bot')
+const masterID = '127060142935113728'
+
+bot.checkRole = async (msg, roleArr) => {
+  var returnobj;
```
Comment (milesflo) [inline src/constants/rolechecker.js]: In JS, variables are almost always camel-case. So in the case of this one it should be
`var returnObj;`

File: src/constants/rolechecker.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  var authorroles = [];
+  var rolesStr = "";
+    for (let i = roleArr.length - 1; i >= 0; i--) {
+      if (await msg.guild.roles.find('name', roleArr[i]) != undefined) {
+        let foundRole = msg.guild.roles.find('name', roleArr[i]);
+        if (await msg.member.roles.has(foundRole.id)) {
+          authorroles.push(" "+foundRole.name)
```
Comment (milesflo) [inline src/constants/rolechecker.js]: Just push the name, not with the space

File: src/constants/rolechecker.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        }
+      }
+    }
+
+    if(authorroles.length > 0) { 
+      returnobj = true;
+      rolesStr = rolesStr + authorroles + " ";
```
Comment (milesflo) [inline src/constants/rolechecker.js]: Use the array.join method here...
`array.join(' ')`

File: src/constants/rolechecker.js (hunk 1 above)
Comment (milesflo) [reply src/constants/rolechecker.js]: And again here, this is going to be a boolean so call it something like `hasRole`

**Review by milesflo — CHANGES_REQUESTED** [review]

**Comment by sergeme** [comment]
implemented the changes, should be OK now. file bot.js has been updated with last push to this branch.

**Review by milesflo — APPROVED** [review]

**Comment by milesflo** [comment]
Great!

**Pull request merged**